src/core.jl

```diff
@@ -94,14 +94,11 @@ function Base.run(x::JSONRPCEndpoint)
             write_transport_layer(x.pipe_out, msg)
         end
     catch err
-        if err isa JSONRPCShutdownException
+        bt = catch_backtrace()
+        if x.err_handler !== nothing
+            x.err_handler(err, bt)
         else
-            bt = catch_backtrace()
-            if x.err_handler !== nothing
-                x.err_handler(err, bt)
-            else
-                Base.display_error(stderr, err, bt)
-            end
+            Base.display_error(stderr, err, bt)
```

Comment:
Does this mean we show an error every time `close` is called on the endpoint?

Reply:
No, there won't be an error thrown if the channel is closed, the `for` loop will just end running and we won't enter the error handler. Closing the endpoint closes the channel.